d/step2_eval.d

```diff
@@ -16,28 +16,24 @@ MalType READ(string str)
 
 MalType eval_ast(MalType ast, Env env)
 {
-    if (typeid(ast) == typeid(MalSymbol))
+    if (MalSymbol sym = cast(MalSymbol)ast)
```

Comment:
should you use `auto sym = ...` here like you did in the other cases?

Reply:
Yes, absolutely. I tried to stick to the original code here. Better to use `auto`.